.github/workflows/ci.yml

```
@@ -56,6 +56,14 @@ jobs:
         run: echo 'memory_limit=-1' >> /usr/local/etc/php/php.ini
       - name: Install Composer
         run: wget -qO - https://raw.githubusercontent.com/composer/getcomposer.org/$COMPOSER_INSTALLER_COMMIT/web/installer | php -- --install-dir=/usr/local/bin --filename=composer --quiet
+      - name: Cache Composer packages
+        uses: actions/cache@preview
+        with:
+          path: ~/.composer/cache/files
+          key: composer-php${{ matrix.php }}-${{ github.sha }}
```

You might want to take a look at https://github.com/localheinz/php-library-template/pull/131, perhaps this is better:

```suggestion
          key: composer-php${{ matrix.php }}-${{ hashFiles('**/composer.json') }}
```

---

That is wrong. It'll prevent the cache from being saved as long as composer.json doesn't change.

---

Works fine for me.

---

@localheinz are you sure? do you mean using `composer.lock` works fine for you?